// aes-gcm/lib/aes-gcm.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var AesGcmModule_1;
const common_1 = require("@nestjs/common");
const _1 = require("./");
let AesGcmModule = AesGcmModule_1 = class AesGcmModule {
    static forRoot(secretKey) {
        return {
            module: AesGcmModule_1,
            providers: [
                {
                    provide: 'AES_SECRET_KEY',
                    useValue: secretKey,
                },
                _1.AesGcmService,
            ],
            exports: [
                'AES_SECRET_KEY',
                _1.AesGcmService,
            ],
        };
    }
};
AesGcmModule = AesGcmModule_1 = __decorate([
    common_1.Global(),
    common_1.Module({})
], AesGcmModule);
exports.AesGcmModule = AesGcmModule;
